### ISSUE TITLE
Issues with DbalLogger on version 2.7

### ISSUE THREAD

**Issue by jbnemanja** [doctrine/DoctrineBundle#1585]
I'm getting a deprecation warning:

```
User Deprecated: The "Symfony\Bridge\Doctrine\Logger\DbalLogger" class implements "Doctrine\DBAL\Logging\SQLLogger" that is deprecated Use {@see \Doctrine\DBAL\Logging\Middleware} or implement {@see \Doctrine\DBAL\Driver\Middleware} instead.
```

I saw the issue was brought up in #1431 and resolved in #1517 for v2.7. But I'm still getting the warning, these are my dependencies

```
"require": {
    "php": ">=8.0",
    "ext-ctype": "*",
    "ext-iconv": "*",
    "doctrine/annotations": "^1.13",
    "doctrine/doctrine-bundle": "^2.7",
    "doctrine/doctrine-migrations-bundle": "^3.2",
    "doctrine/orm": "^2.13",
    "gedmo/doctrine-extensions": "^3.10",
    "gesdinet/jwt-refresh-token-bundle": "^1.1",
    "lexik/jwt-authentication-bundle": "^2.16",
    "sensio/framework-extra-bundle": "^6.2",
    "stof/doctrine-extensions-bundle": "^1.7",
    "symfony/console": "5.4.*",
    "symfony/dotenv": "5.4.*",
    "symfony/flex": "^1.17|^2",
    "symfony/framework-bundle": "5.4.*",
    "symfony/http-client": "5.4.*",
    "symfony/http-foundation": "5.4.*",
    "symfony/property-access": "5.4.*",
    "symfony/proxy-manager-bridge": "5.4.*",
    "symfony/runtime": "5.4.*",
    "symfony/security-bundle": "5.4.*",
    "symfony/serializer": "5.4.*",
    "symfony/uid": "5.4.*",
    "symfony/validator": "5.4.*",
    "symfony/yaml": "5.4.*"
},
"require-dev": {
    "friendsofphp/php-cs-fixer": "^3.12",
    "phpstan/phpstan": "^1.8",
    "symfony/stopwatch": "5.4.*",
    "symfony/web-profiler-bundle": "5.4.*",
    "vimeo/psalm": "^4.29"
},
```

I'm using PHP 8.1 and Symfony 5.4

**Comment by dmaicher** [comment]
Yeah indeed. I also see this deprecation and it seems we are not removing the `doctrine.dbal.logger` service which is using that class. I will have a look at fixing this on 2.8.x where we dropped DBAL 2 support completely.

**Comment by dmaicher** [comment]
This was fixed on 2.8.x in https://github.com/doctrine/DoctrineBundle/pull/1587
